crypto3/libs/zk/include/nil/crypto3/zk/snark/systems/plonk/placeholder/preprocessor.hpp

```diff
@@ -580,15 +606,7 @@ namespace nil {
                         q_last_q_blind[0] = lagrange_polynomial(basic_domain, usable_rows);
                         q_last_q_blind[1] = selector_blind(usable_rows, basic_domain);
 
-                        plonk_public_polynomial_dfs_table<FieldType>
-                            public_polynomial_table =
-                                plonk_public_polynomial_dfs_table<FieldType>(
-                                    detail::column_range_polynomial_dfs<FieldType>(public_assignment.move_public_inputs(),

```

Comment:
this place was essentially the way to invalidate pointer to public table (valid shared_ptr may be accessible from the owning plonk_table while the actual content is already moved out)

And as fas as these vectors are only used to build new polynomial_dfs objects - it is more reasonable to pass them as const& and then just release the owning shared_ptr